### PR TITLE
feat: add better comms subscription handling

### PR DESCRIPTION
**Pull request by nduchak**
- add GC for room handlers
 - create an on-message listener for each request

### CODE REVIEW

File: src/communication/index.ts [hunk 1]
```diff
@@ -33,39 +44,30 @@ export function subscribeForOffer (
...
 
     roomLogger.debug(`Receive message: ${JSON.stringify(message)}`)
 
-    if (from !== rooms.get(topic)?.peerId) {
+    if (from !== peerId) {
       return
     }
 
-    await handler(message as CommsMessage<CommsPayloads>)
+    if (await handler(message as CommsMessage<CommsPayloads>)) {
```
Comment (AuHau) [inline src/communication/index.ts:63]:
```suggestion
	// If handler returns true, than it signals no further messages are expected
	// and we remove this handler.
    if (await handler(message as CommsMessage<CommsPayloads>)) {
```

File: src/communication/handler.ts [hunk 2]
```diff
@@ -38,28 +38,20 @@ async function hashPinnedHandler (
     await storageProvider?.rm(job.fileHash)
     roomLogger.info(`File ${job.fileHash} for offer ${job.offerId} unpinned`)
   }
-
-  // Leave room if not active jobs for that offer
-  const pendingJobsForOffer = await UploadJob.count({ where: { offerId } })
```
Comment (AuHau) [inline src/communication/handler.ts:43]: Hmmm without this we will stay forever connected to the Room though 🤔
It could stay here right?

**Review by AuHau — CHANGES_REQUESTED** [review]

File: src/communication/handler.ts (hunk 2 above)
Comment (nduchak) [reply src/communication/handler.ts:43]: We had the same logic in the GC function, but here we have a chance to leave the channel faster

Comment (nduchak) [reply src/communication/handler.ts:43]: Without waiting until GC is executed

File: src/communication/index.ts (hunk 1 above)
Comment (nduchak) [reply src/communication/index.ts:63]: Done

File: src/communication/handler.ts (hunk 2 above)
Comment (nduchak) [reply src/communication/handler.ts:43]: Probably i will revert this

**Review by AuHau — APPROVED** [review]
LGTM!

**Pull request merged**